### ISSUE TITLE
bamtools stats files not recognized anymore version > 1.5

### ISSUE THREAD

**Issue by alexandre-nadin** [ewels/MultiQC#1023]
**Description of bug:**
From version >= 1.6, multiqc doesn't recognize bam stats files anymore, even when explicitely fed to the command. In the example below, I show that with the same bam stat files, version 1.5 recognizes and processes them, while upper version 1.7 does not.

**MultiQC Error log: **
```
# Version 1.7
$ multiqc -o ./ .*stats
[INFO   ]         multiqc : This is MultiQC v1.7 (fbfcde9)
[INFO   ]         multiqc : Template    : default
[INFO   ]         multiqc : Searching '01.bam.stats'
[INFO   ]         multiqc : Searching '01.bamstats'
[INFO   ]         multiqc : Searching 'bam.stats'
Searching 3 files..  [####################################]  100%
[WARNING]         multiqc : No analysis results found. Cleaning up..
[INFO   ]         multiqc : MultiQC complete

# Version 1.5
$ multiqc -o ./ *stats
[WARNING]         multiqc : MultiQC Version v1.7 now available!
[INFO   ]         multiqc : This is MultiQC v1.5.dev0 (fbfcde9)
[INFO   ]         multiqc : Template    : default
[INFO   ]         multiqc : Searching 'bam.stats'
[INFO   ]         multiqc : Searching '01.bamstats'
[INFO   ]         multiqc : Searching '01.bam.stats'
[INFO   ]        bamtools : Found 2 bamtools stats reports
[INFO   ]         multiqc : Compressing plot data
[INFO   ]         multiqc : Report      : multiqc_report.html
[INFO   ]         multiqc : Data        : multiqc_data

```

**MultiQC run details (please complete the following):**
 - Command used to run MultiQC: `multiqc .`
 - MultiQC Version: v1.7
 - Operating System: REdhat 2.6.32-279.2.1.el6.x86_64 GNU/Linux
 - Python Version: Python 3.6.7
 - Method of MultiQC installation: conda

 **Additional context**

**Comment by ewels** [comment]
Thanks for reporting this @alexandre-nadin - do you have an example file that you could attach for me to debug with? The Travis CI tests are [still picking up the files](https://travis-ci.org/ewels/MultiQC/jobs/594469863#L2098) from the [test data repo](https://github.com/ewels/MultiQC_TestData/tree/master/data/modules/bamtools/stats), so it must be something quite subtle..

Phil

**Comment by alexandre-nadin** [comment]
Here is a reproducible bash script for both versions 1.5 and 1.7 of multiqc, installed with conda:

```
#!/usr/bin/env bash
for v in 1.5 1.7; do
  conda create -n mqc${v} multiqc=${v} -y &> mqc${v}_create.log
  conda activate mqc${v}
  LC_ALL=en_US.utf8 LANG=en_US.utf8 \
    multiqc -f sample.bamstats      \
     &> mqc${v}.log
  conda deactivate
  conda uninstall -n mqc${v} --all -y &> mqc${v}_uninstall.log
done;
```

With log for v1.5, the bamstat file was recognized:
```
$ cat mqc1.5.log 
/lustre1/ctgb-usr/local/miniconda3/envs/mqc1.5/lib/python2.7/site-packages/multiqc/utils/config.py:45: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.
  configs = yaml.load(f)
/lustre1/ctgb-usr/local/miniconda3/envs/mqc1.5/lib/python2.7/site-packages/multiqc/utils/config.py:51: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.
  sp = yaml.load(f)
[WARNING]         multiqc : MultiQC Version v1.7 now available!
[INFO   ]         multiqc : This is MultiQC v1.5 (fbfcde9)
[INFO   ]         multiqc : Template    : default
[INFO   ]         multiqc : Searching 'sample.bamstats'
[INFO   ]        bamtools : Found 1 bamtools stats reports
[INFO   ]         multiqc : Compressing plot data
[INFO   ]         multiqc : Report      : multiqc_report.html
[INFO   ]         multiqc : Data        : multiqc_data
[INFO   ]         multiqc : MultiQC complete
```

With v1.7, the bamstat was not recognized:
```
$ cat mqc1.7.log 
/lustre1/ctgb-usr/local/miniconda3/envs/mqc1.7/lib/python3.6/site-packages/multiqc/utils/config.py:45: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.
  configs = yaml.load(f)
/lustre1/ctgb-usr/local/miniconda3/envs/mqc1.7/lib/python3.6/site-packages/multiqc/utils/config.py:51: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.
  sp = yaml.load(f)
[INFO   ]         multiqc : This is MultiQC v1.7 (fbfcde9)
[INFO   ]         multiqc : Template    : default
[INFO   ]         multiqc : Searching 'sample.bamstats'
[WARNING]         multiqc : No analysis results found. Cleaning up..
[INFO   ]         multiqc : MultiQC complete
``` 

Here is the content of the bamstats file:
```
$ cat sample.bamstats 

**********************************************
Stats for BAM file(s): 
**********************************************

Total reads:       7956361
Mapped reads:      7780129	(97.785%)
Forward strand:    3337363	(41.9458%)
Reverse strand:    4618998	(58.0542%)
Failed QC:         0	(0%)
Duplicates:        0	(0%)
Paired-end reads:  0	(0%)


```

You can also download it here (once): https://file.io/pPb4Gy

Thanks,
Alex

**Comment by ewels** [comment]
Hi @alexandre-nadin,

Apologies for the delay, just coming back to this. Despite your excellent detailed response, I'm struggling to reproduce the problem. The download has expired sadly, but if I paste your `sample.bamstats` file then MultiQC (v.18dev) finds it and parses it without any problem.

It's tricky to tell from pastes, but does your file have a blank line at the top? If so then this is probably a duplicate of #978 and already fixed in v1.8dev.

If you're able to test your code again with the latest development code that would be great. If it still breaks, please send the files again (you should be able to just attach them to this issue without using another service - zip the files if GitHub complains about file formats).

Phil

**Comment by ewels** [comment]
Closing this issue on the assumption that it's a duplicate and already fixed. If you find that it's still problematic let me know and I will reopen it.
